lib/registration.pm

```diff
@@ -252,9 +252,14 @@ sub cleanup_registration {
     register_product();
 
 Wrapper for SUSEConnect -r <regcode>. Requires SCC_REGCODE variable.
+SUSEConnect --url with SMT/RMT server.
 =cut
 sub register_product {
-    assert_script_run('SUSEConnect -r ' . get_required_var('SCC_REGCODE'), 200);
+    if (get_var('SMT_URL')) {
```

Comment:
How do you think to add a parameter for the function to add extra parameter for SUSEConnect, so later we can use this parameter to extend other special scenarios. WDYT?

Reply:
For other SUSEConnect register moudles,we can use ''add_suseconnect_product''. For register product, we only have three methods: scc, smt and rmt. So I think we don't need add parameter at here.